src/Console/ConfigurationResolver.php

```diff
@@ -406,38 +406,26 @@ private function resolveConfigPath()
         // - when we don't have valid finder - prepare new iterator
         $iterator = null;
         $currentFinder = $this->config->getFinder();
+        $path = $this->path;
+        $isFile = is_file($path);
 
         try {
-            $currentIteratorItems = array_map(
-                function (\SplFileInfo $current) {
-                    return $current->getRealPath();
-                },
-                iterator_to_array($currentFinder, false)
-            );
-
-            $nestedIterator = is_file($this->path)
-                ? new \ArrayIterator(array(new SymfonySplFileInfo(
-                    $this->path,
-                    substr(dirname($this->path), strlen($this->cwd) + 1),
-                    substr($this->path, strlen($this->cwd) + 1)
-                )))
-                : Finder::create()->in($this->path)->getIterator()
-            ;
-
             $iterator = new \CallbackFilterIterator(
-                $nestedIterator,
-                function (\SplFileInfo $current) use ($currentIteratorItems) {
-                    return in_array($current->getRealPath(), $currentIteratorItems, true);
+                $currentFinder instanceof \IteratorAggregate ? $currentFinder->getIterator() : $currentFinder,
+                function (\SplFileInfo $current) use ($path, $isFile) {
+                    return $isFile
```

Comment:
I thought about it, too. But I guess there is no real performance impact, and imho the other solution is more readable.
But can change it, if you want.